### PR TITLE
Add alias cert mode support

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #2056

Add `is_device_cert_model` to set different `basicConstraints`;

### CODE REVIEW

File: os_stub/cryptlib_openssl/pk/x509.c [hunk 1]
```diff
@@ -2412,17 +2415,18 @@ bool libspdm_set_attribute_for_req(X509_REQ *req, uint8_t *req_info, size_t req_
...
+ * @param[in, out]      csr_pointer           For input, csr_pointer is buffer address to store CSR.
+ *                                            For output, csr_pointer is address for stored CSR.
+ *                                            The csr_pointer address will be changed.
  *
  * @retval  true   Success.
  * @retval  false  Failed to gen CSR.
  **/
 bool libspdm_gen_x509_csr(size_t hash_nid, size_t asym_nid,
                           uint8_t *requester_info, size_t requester_info_length,
+                          bool is_device_cert_model,
```
Comment (jyao1) [inline os_stub/cryptlib_openssl/pk/x509.c:2429]: I dont think we can use is_device_cert_model in crypto_lib.

File: os_stub/cryptlib_mbedtls/pk/x509.c [hunk 2]
```diff
@@ -1702,6 +1703,8 @@ bool libspdm_set_attribute_for_req(mbedtls_x509write_csr *req, uint8_t *req_info
  * @param[in]      requester_info        requester info to gen CSR
  * @param[in]      requester_info_length The len of requester info
  *
+ * @param[in]       is_ca                if ture,set basic_constraints ture; Otherwise, set to false.
```
Comment (Zhiqiang520) [inline os_stub/cryptlib_mbedtls/pk/x509.c:1706]: here change to `if ture, set basic_constraints CA: ture`,  more accurate.

File: os_stub/cryptlib_null/pk/x509.c [hunk 3]
```diff
@@ -748,6 +748,8 @@ int32_t libspdm_x509_compare_date_time(const void *date_time1, const void *date_
  * @param[in]      requester_info        requester info to gen CSR
  * @param[in]      requester_info_length The len of requester info
  *
+ * @param[in]       is_ca                if ture,set basic_constraints ture; Otherwise, set to false.
```
Comment (Zhiqiang520) [inline os_stub/cryptlib_null/pk/x509.c:751]: here change to ' if **true**, set basic_constraints: **CA**: ture ', more accurate.

File: os_stub/cryptlib_openssl/pk/x509.c [hunk 4]
```diff
@@ -2401,6 +2401,8 @@ bool libspdm_set_attribute_for_req(X509_REQ *req, uint8_t *req_info, size_t req_
  * @param[in]      requester_info        requester info to gen CSR
  * @param[in]      requester_info_length The len of requester info
  *
+ * @param[in]       is_ca                if ture,set basic_constraints ture; Otherwise, set to false.
```
Comment (Zhiqiang520) [inline os_stub/cryptlib_openssl/pk/x509.c:2404]: here change to ' if **true**, set basic_constraints: **CA**: ture ', more accurate.

File: unit_test/test_spdm_responder/csr.c [hunk 5]
```diff
@@ -961,6 +1090,84 @@ void libspdm_test_responder_csr_case10(void **state)
...
+        m_libspdm_use_asym_algo;
+
+    /*set alias cert mode*/
+    spdm_context->local_context.capability.flags |=
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ALIAS_CERT_CAP;
+    is_device_cert_model = false;
+    if((spdm_context->local_context.capability.flags &
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ALIAS_CERT_CAP) == 0) {
+        is_device_cert_model = true;
+    }
```
Comment (Zhiqiang520) [inline unit_test/test_spdm_responder/csr.c:1134]: here `libspdm_test_responder_csr_case11` is `with alias_cert mode`.
I think we already set the test scenario explicitly at the line 1128 and 1130, so remove the above line 1131 and 1134.

File: os_stub/cryptlib_mbedtls/pk/x509.c (hunk 2 above)
Comment (Wenxing-hou) [reply os_stub/cryptlib_mbedtls/pk/x509.c]: Thanks. Changed.

File: os_stub/cryptlib_openssl/pk/x509.c (hunk 1 above)
Comment (Wenxing-hou) [reply os_stub/cryptlib_openssl/pk/x509.c]: Thanks. I use the `is_ca` to replace `is_device_cert_model` in crypto_lib.

File: unit_test/test_spdm_responder/csr.c [hunk 6]
```diff
@@ -176,6 +295,11 @@ void libspdm_test_responder_csr_case1(void **state)
     spdm_context->connection_info.algorithm.base_asym_algo =
         m_libspdm_use_asym_algo;
 
+    is_device_cert_model = false;
+    if((spdm_context->local_context.capability.flags &
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ALIAS_CERT_CAP) == 0) {
+        is_device_cert_model = true;
+    }
```
[the remark this answers is not in the document]
Comment (Wenxing-hou) [reply unit_test/test_spdm_responder/csr.c]: Thanks. That make sense. I have changed the code.

File: unit_test/test_spdm_responder/csr.c (hunk 5 above)
Comment (Wenxing-hou) [reply unit_test/test_spdm_responder/csr.c]: Thanks. I have changed the code.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**